### DISCUSSION TITLE
self closing twig blocks as HTML components

### DISCUSSION THREAD

**Discussion by tacman** [symfony/symfony#51597] [Q&A]
I love the new twig block syntax as an HTML component.

```html
<twig:block name="body">
    <twig:block name="title">Page Title</twig:block>
   {% block subtitle 'Page subtitle' %}   
</twig:block>
```

I like to use the short version of block with traditional twig, as in the example above, but I don't think there's a way to do that.

Poking around the code, I see "content" is the keyword to forward block contents: 

https://symfony.com/bundles/ux-twig-component/current/index.html#passing-content-blocks-to-components

So I was thinking about a syntax for self-closing blocks, e.g.

```html
<twig:block name="title" content="My Page Title" />
<twig:block name="title" :content="company.name" />
```

Thoughts?  Is there already a way to do this?

Thanks,

**Comment by chapterjason** [comment]
Looks like there is already something like this: https://github.com/symfony/ux/blob/0950bfa271d655f68250a4f9e48ab88ae51bb73c/src/TwigComponent/src/Twig/TwigPreLexer.php#L102

**Reply by tacman** [reply]
Yes, but that only works without a content, e.g. <twig:Alert type='danger' msg="watch out!" />  My issue was a way to pass content that is inside the block.
